### PR TITLE
chore: reduce dowload of go mods

### PR DESCRIPTION
**Pull request by RafilxTenfen**
First initial improvement in CI e2e test

Next steps is to cache docker images and split execution of e2e test

### CODE REVIEW

**Review by KonradStaniec — APPROVED** [review]
LGTM! (the e2e tests are passing on CI which for me is most important part)

Though lets wait for feedback from @filippos47 or @liam-icheng-lai  before merging as this:
- ditches usage of our re-usable templates 
- changes babylon docker image

File: contrib/images/babylond/Dockerfile
Comment (RafilxTenfen) [reply contrib/images/babylond/Dockerfile:1]: ops, this part can be removed, was testing something

File: .github/workflows/ci.yml
Comment (RafilxTenfen) [reply .github/workflows/ci.yml:88]: my bad, it would use the cache if we didn't run the `make test-e2e-cache` with sudo, had a fix for this at #16

File: contrib/images/babylond/Dockerfile
Comment (RafilxTenfen) [reply contrib/images/babylond/Dockerfile:55]: not necessary, but I can leave if you prefer

File: contrib/images/babylond/Dockerfile
Comment (RafilxTenfen) [reply contrib/images/babylond/Dockerfile]: removed unnecessary multistage https://github.com/babylonlabs-io/babylon/pull/15/commits/f34b3d3622216fae2018992dac255577ee2956ef

**Review by filippos47 — COMMENTED** [review]
What's the reason behind removing the reusable pipelines?
We can adapt them to accommodate any babylon-specific use case.

**Comment by RafilxTenfen** [comment]
> What's the reason behind removing the reusable pipelines? We can adapt them to accommodate any babylon-specific use case.

For this PR we could continue to use the template, but the next one that I am working (https://github.com/babylonlabs-io/babylon/pull/16) it is very specific for babylon
maybe we could continue to use the template for all and only do specific for e2e testing

@filippos47

**Comment by filippos47** [comment]
>  Maybe we could continue to use the template for all and only do specific for e2e testing

I think that's a good solution for now.

**Comment by RafilxTenfen** [comment]
closed in favor of https://github.com/babylonlabs-io/babylon/pull/16

@filippos47 @KonradStaniec

**Pull request closed without merging**
